### ISSUE TITLE
Add a tool to export rasters and the Lizmap canvas as raster/image

### ISSUE THREAD

**Issue by gioman** [3liz/lizmap-web-client#46]
Just an idea for a eventual future addition, as it is available on other webgis frameworks and it seems to be very popular: allow export the canvas as georeferenced image (for example as png + worldfile) as also raster layers in a similar way as is already possible for vectors.

Of course the latter would have to be a download of the original dataset, not the rendered version as WMS, so there could be some limitations (i.e. external WMS layers) but other than that the original raster layer (probably a file based one) should be "passed to" the final user, this way we could take advantage of LMWC functionalities such as global/layer/project permissions.

**Comment by igeofr** [comment]
May be easier with the latest version of OpenLayers. 
[An example](https://bl.ocks.org/ThomasG77/415906d5ac34bf710f8c2fbe42f48201)

**Comment by gioman** [comment]
Reiterating the importance of this functionality, as seen in many other WebGIS platforms (see below). Still waiting for a contact to try see how this could be materialized.






![Screenshot 2023-06-02 140322](https://github.com/3liz/lizmap-web-client/assets/1951107/5787f845-62b5-4b94-a86e-23ca7a3858ea)

**Comment by gioman** [comment]
> @Gustry: on a other topic, you suggest to use the "media folder" as a workaround. It could be a kink of solution, the problem is to allowed the download only to "admin user" and not to "anonymous" or other "normal users"!

@katagen you can create a link pointing to anything in "media", if the project is accessible to anonymous, the file will be also be accessible... I guess.

**Comment by katagen** [comment]
I know, and it is the problem. For vector layers, when you set the rights, you can specify that only "admin" can download the layers.
But if I use the media folder with a link, everyone can download the file... which I don't want!

**Comment by Gustry** [comment]
Yes, I totally agree, that's why I said it was a "quick workaround", if this solution was OK.
Yes, ACL are not checked, so it can be solution.

I have rephrased the label in #4747, downloading raster layer have never been implemented in Lizmap. Feel free to contact us.

**Comment by gioman** [comment]
> But if I use the media folder with a link, everyone can download the file... which I don't want!

@katagen you can configure layer to be visible to specific LM groups, so you could have the same layer published twice, one visible only to anonymous(*) and one only to users belonging to a specific group, where only the latter would have the link for direct download.

(*) unfortunately I think this is not really doable because the lat time I tried/checked there was no way to use "anonymous" as group name in LM plugin to set some type of permission to users that are not logged. I'm sure I filed a ticket about this problem.

**Comment by gioman** [comment]
> Feel free to contact us

@Gustry not sure if it was clear but here https://github.com/3liz/lizmap-web-client/issues/46#issuecomment-1476697611 I had sent a signal in this sense...

**Comment by Gustry** [comment]
Indeed @gioman indeed... :/
There is like [260 tickets tagged 'enhancement'](https://github.com/3liz/lizmap-web-client/labels/enhancement), GitHub doesn't make this kind of tracking easy. Let us check this ticket.
